Draft to see commits since stable

anjakefala opened the pull request.

anjakefala committed:
[dev] bump visidata to v3.3dev

anjakefala committed:
Merge branch 'stable' into develop

chenrui333 committed:
fix: include missing files in MANIFEST.in

Signed-off-by: Rui Chen <rui@chenrui.dev>

midichef committed:
[basesheet-] start new column names fresh for each sheet

And fix undo of addcol-incr addition, so it decrements the name counter.

midichef committed:
[save-] suggest 'stdin' as filename base instead of '-'

midichef committed:
[plugins] liveupdate, fix broken addcol-new

And assign a name to the column on creation, to avoid incrementing the
sheet's column-naming counter.

midichef committed:
[sheets-] prevent 80x25 layout after redraw

Prevents checkCursor from making layout changes, in the
interval between vd.redraw() and vd.draw_all() where _scr is None.

midichef committed:
[column-] simplify getMaxWidth

Its behavior is not changed at all.

midichef committed:
[column-] improve getMaxWidth values for dict/list/tuple

For collection columns having width None or 0, getMaxWidth would
return just the width of the column name. For other collection col
widths, getMaxWidth would return 2*col.width+2.

midichef committed:
[column-] stop getCell width-doubling for dict/list/tuple

midichef committed:
[column-] document the clipping of dict/list/tuple for formatting

anjakefala committed:
[sheet-] fix reload on sheets without col layout  #2790

anjakefala committed:
Merge pull request #2789 from chenrui333/fix/manifest-missing-files

fix: include missing files in MANIFEST.in

midichef committed:
[sheets- undo-] revert multiple commits affecting cache and layout

Revert "[undo-] recalc column layout after undo and contract_cols #2607"
This reverts commit f9005368d357e69bacc46c00e8d6bd017f900181.

Revert "[reload-] re-calculate col layout before reload"
This reverts commit 4fe7e764efe13ae3ff8322912fec0735088ed56c.

Revert "[reload-] tolerate stale columns in drawcache during allAggregators #2607 #2763"
This reverts commit 04de60dfed76dd2218ed4d97565317ee3d060d1c.

Revert "[sheet-] fix reload on sheets without col layout  #2790"
This reverts commit bb85182bd09be1ebf26240ff474c563b6d06fa01.

midichef committed:
[sheets-] calc col visibility before using nScreenRows

prevents IndexError on availCols, after undo or reload operations that
decrease the number of columns

midichef committed:
[sheets-] remove calcSingleColLayout use of inaccurate visibleRows

A very minor improvement to display. It will make column layout more
consistent immediately after operations that change the number of
columns with aggregators.

midichef committed:
[sheets-] match width returned by calcSingleColLayout to width assigned

Previously they could differ when the rightmost column extended
past the right side of the screen. But the difference caused
no problems.

midichef committed:
[sheets-] rework checkCursor visible col layout

add comments clarifying intention,
replace 'while True' with more robust condition to avoid infinite loop,
simplify code using mincolidx and maxcolidx that had no effect

midichef committed:
[features-] hide-uniform-cols: remove cache clear

Because it is now unnecessary. It may have been unnecessary
when it was first introduced.

midichef committed:
[sheets-] remove calcColLayout from cursorRight

it is redundant now that cursorCheck runs calcColLayout

midichef committed:
[sheets-] drop offscreen col from _visibleColLayout

When the right side of the screen contained a column separator
or the last character of a column, _visibleColLayout would
incorrectly include the offscreen column to the right.

The inclusion was mostly harmless, as most retrievals from
_visibleColLayout check the x coordinate of the column for
validity. But allAggregators does not check it, so aggregators
for that offscreen column were shown when they should not
have been visible.

And if disp_column_sep was '', two such offscreen columns were
included, instead of just one.

midichef committed:
[tests-] test.sh: exit non-zero if output differs

midichef committed:
[tests-] skip sysedit commands

because sysedit-cell blocks further tests until user quits,
when $EDITOR is vi

midichef committed:
[tests-] update ColumnsSheet to have widths immediately

midichef committed:
[test-] allow parallel tests in test.sh with -j

midichef committed:
[build-] skip slow installation of unused apt packages

Sometimes install took 12 seconds, other times > 60.

midichef committed:
[test-] silence shellcheck errors/warnings for test.sh

They would only make a difference for filenames with spaces,
which we did not have.

saulpw committed:
[status] fix error when no help_sidebars

anjakefala committed:
Merge pull request #2811 from midichef/streamline_tests

streamline the testing process

anjakefala committed:
Merge pull request #2800 from midichef/save_from_stdin

[save-] suggest 'stdin' as filename base instead of '-'

anjakefala committed:
Merge pull request #2803 from midichef/redraw_not_80x25

[sheets-] prevent 80x25 layout after redraw

midichef committed:
[cliptext-] restore display translation of unprintable chars

anjakefala committed:
[resize-] fix resize-col-half for new columns with no rows  #2795

midichef committed:
[form-] after confirm, Esc/^C/^Q/q show disconfirm msg

midichef committed:
[build-] try simultaneous jobs in test.sh: 4 jobs

midichef committed:
[threads-] add asyncsingle_queue decorator

midichef committed:
[reload-] wait for any previous reload_rows to finish #2808

midichef committed:
[save-] add threading warnings to saveSheets docstring

midichef committed:
[macro-] fix corruption when saving macros after deletion

midichef committed:
[palette-] scroll choices with PageUp/PageDown

midichef committed:
[palette-] hide choices that do not match input

midichef committed:
[input-] add guard to draw() near right edge

midichef committed:
[input-] handle screen resize during input()

midichef committed:
[input-] handle screen resize during editCell()

midichef committed:
[palette-] keep case of results, allow case-sensitive search

midichef committed:
[sidebar-] cache formatter_helpstr for each sheet

Speeds up display significantly, by 7 milliseconds per command.

dennisangemi committed:
feat: auto-detect CSV format when csv_delimiter is specified

When user specifies --csv-delimiter with a non-default value, automatically
detect the file as CSV format instead of requiring -f csv flag.

This prevents the 'unknown filetype' error and improves user experience
when working with CSV files that use non-comma delimiters.

saulpw committed:
Update csv.py

Use options.getdefault() instead of hardcoding default delimiter.

midichef committed:
[tar-] add cols for ext and file type desc, use name as keycol

midichef committed:
[sheets-] on reload, fetch fresh data for URLs

Co-authored-by: anjakefala <anja.kefala@gmail.com>

midichef committed:
[palette-] add instruction to use Esc to cancel

anjakefala committed:
Merge pull request #2834 from midichef/tar_columns

[tar-] add cols for ext and file type desc, use name as keycol

anjakefala committed:
Merge pull request #2830 from midichef/cache_formatter_helpstr

speed up display by caching formatter_helpstr

anjakefala committed:
Merge pull request #2832 from dennisangemi/develop

feat: auto-detect CSV format when csv_delimiter is specified

midichef committed:
[archive-] fix loading zip files from inside archive files

Previously, trying to open "test.zip" from inside a .tar or .zip,
would only search for "test.zip" as a file in the current
directory, instead of reading from the archive file.

midichef committed:
[archive-] fix error extracting to overwrite existing file

midichef committed:
[archive-] fix extracting to a chosen path

Previously, it had been extracting to the current directory.

midichef committed:
[cliptext-] fix display of empty markup as several blank lines

Empty markup like "[:error][/]" would become 5 blank lines.
Text of "[:]" or "[/]" would become 3 blank lines.

anjakefala committed:
Merge pull request #2837 from midichef/archive_paths

fix path problems in archive files

anjakefala committed:
Merge pull request #2835 from midichef/http_reload

[sheets-] on reload, fetch fresh data for URLs

anjakefala committed:
Merge pull request #2822 from midichef/palette_scroll

[palette-] scroll matches with PageUp/PageDown

anjakefala committed:
Merge pull request #2817 from midichef/test_parallel

[test-] allow parallel tests in test.sh with -j

anjakefala committed:
Merge pull request #2818 from midichef/reload_rows_single

[reload] run reload_rows one at a time

anjakefala committed:
Merge pull request #2819 from midichef/macro_save

[macro] fix macro corruption when saving via putChanges

anjakefala committed:
[test] add cliptext test for display of unprintable characters  #2815

anjakefala committed:
Merge pull request #2815 from midichef/translate_unprintable

prevent output of raw characters to terminal

midichef committed:
fix plugin autoload for Python versions < 3.10

midichef committed:
[input-] handle screen resize during inputMultiple()

midichef committed:
[input-] if input keeps restarting, error instead of fail

anjakefala committed:
Merge pull request #2843 from midichef/importlib_v3_8_v3_9

fix plugin autoload for Python versions < 3.10

rsmirnov90 committed:
Bugfix to fix Visidata crash on startup on NetBSD (and probably other *BSD) wscons tty consoles, which do not expose "mousemask" in their system curses implementations.

Signed-off-by: Russ <russ@sentibyte.com>

midichef committed:
fix editText for files on command line with -b -i -p (#2840)

* [main-] fully unset batch mode when interactive

The 'batch' option was set for files that were arguments on the
command line. That caused all edits on those sheets to return
empty strings.

* [tests] add 2840 to manual tests checklist

---------

Co-authored-by: anjakefala <anja.kefala@gmail.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/saulpw/visidata?pullRequest=2850) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/saulpw/visidata?pullRequest=2850) before we can accept your contribution.<br/>**5** out of **6** committers have signed the CLA.<br/><br/>:white_check_mark: anjakefala<br/>:white_check_mark: chenrui333<br/>:white_check_mark: dennisangemi<br/>:white_check_mark: saulpw<br/>:white_check_mark: midichef<br/>:x: rsmirnov90<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/saulpw/visidata?pullRequest=2850) it.</sub>

saulpw committed:
[disp_help_flags] added options.disp_help_flags; deprecated options.disp_expert

- these feature flags are supported: cmdpalette hints nometacols guides
  inputkeys inputfield all
- see helpstr for more details
- set option to '' to turn off

midichef committed:
[freqtbl-] stop printing status for selection on source sheet

Because each select runs in its own thread, race conditions made
the displayed line counts very inaccurate for large bins.

saulpw committed:
[help] reorder help_flags helpstr

midichef committed:
[sheets-] move colname counter to from BaseSheet to TableSheet

midichef committed:
[sheets-] simplify column autoname during newSheet

anjakefala committed:
[tests] test auto-naming in new sheet #2855

anjakefala committed:
[dev update CHANGELOG

saulpw committed:
[dev] cleanup changelog

anjakefala committed:
[dev] Bump to v3.3
